### PR TITLE
GH issue 3398 - option to disable doxygen warn as error

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -952,6 +952,13 @@ if (HDF5_BUILD_DOC AND EXISTS "${HDF5_DOXYGEN_DIR}" AND IS_DIRECTORY "${HDF5_DOX
 # check if Doxygen is installed
   find_package(Doxygen)
   if (DOXYGEN_FOUND)
+    option (HDF5_ENABLE_DOXY_WARNINGS "Enable fail if doxygen parsing has warnings." ON)
+    mark_as_advanced (HDF5_USE_FOLDERS)
```
Comment (jhendersonHDF) [inline CMakeLists.txt]: HDF5_ENABLE_DOXY_WARNINGS

File: configure.ac
Comment (jhendersonHDF) [inline configure.ac:1]: Looks like CMake can support FAIL_ON_WARNINGS vs. NO, while Autotols would be NO vs. YES. Is that a big deal? Also, is it a problem if Autotools specifies "no" and "yes" vs "NO" and "YES" to doxygen?

**Comment by byrnHDF** [comment]
Dang, I need a break!

File: configure.ac
Comment (byrnHDF) [reply configure.ac:1]: No it needs the correct text

Comment (byrnHDF) [reply configure.ac:1]: Updated

**Review by jhendersonHDF — APPROVED** [review]

File: configure.ac [hunk 2]
```diff
@@ -1207,6 +1207,30 @@ AC_ARG_ENABLE([doxygen],
...
 AC_MSG_RESULT([$HDF5_DOXYGEN])
 
+## Check if they would like to enable doxygen warnings as errors
+##
+
+## This needs to be exposed for the library info file.
+AC_SUBST([HDF5_DOXY_WARNINGS])
+
+## Default is to not build DOXYGEN
+HDF5_DOXY_WARNINGS=no
```
Comment (derobins) [inline configure.ac]: I think you want DOXY_ERR here. Also, the comment is misleading (copy-paste error?).

File: release_docs/RELEASE.txt [hunk 3]
```diff
@@ -47,6 +47,18 @@ New Features
 
     Configuration:
     -------------
+    - Added new options for CMake and autotools to control the Doxygen
```
Comment (derobins) [inline release_docs/RELEASE.txt]: Autotools is capitalized

File: release_docs/RELEASE.txt [hunk 4]
```diff
@@ -47,6 +47,18 @@ New Features
...
+      warnings as errors setting.
+
+        * HDF5_ENABLE_DOXY_WARNINGS: ON/OFF (Default: ON)
+        * doxygen-errors: enable/disable (Default: enable)
```
Comment (derobins) [inline release_docs/RELEASE.txt]: Should be --enable-doxygen-errors

File: configure.ac (hunk 2 above)
Comment (byrnHDF) [reply configure.ac]: done

File: release_docs/RELEASE.txt (hunk 3 above)
Comment (byrnHDF) [reply release_docs/RELEASE.txt]: done

File: release_docs/RELEASE.txt (hunk 4 above)
Comment (byrnHDF) [reply release_docs/RELEASE.txt]: done

**Review by derobins — APPROVED** [review]

**Pull request merged**